### ISSUE TITLE
stored thumbnail images at different dimensions

### ISSUE THREAD

**Issue by desb42** [gnosygnu/xowa#447]
looking at en.wikipedia.org-file-core.xowa (2019-03-01)

fsdb_thm has (potentially) multiple entries for an image at different file dimensions (and hence different filesizes)
I wanted to check which images had the most different file dimensions and what amount of filespace they used
I wrote this query

    select d.tid, f.fil_name, d.cnt, d.mts "max size", d.sts "tot size", d.sts/d.cnt "avg" from fsdb_fil f
    join 
    (select c.tid, count(*) cnt, max(thm_size) mts, sum(thm_size) sts from
    (select thm_owner_id as tid, thm_id, thm_size from fsdb_thm 
    ) c
    group by c.tid ) d on d.tid=f.fil_id
    where d.cnt > 4
    order by d.cnt desc

With the cutoff of a minimum of 5 different sizes there are 56,683 images
Summing up the count (cnt) gives 391,830 entries in fsdb_thm
total space used 5.47Gb

Would it be reasonable to suggest that only the largest needs to be stored?
That is, always serve the largest file regardless of the other (smaller) dimensions desired
(the browser should scale the image appropriately)

In that case, if only the largest is stored then the space would be 1.65Gb
(in this case a saving of approx 4Gb)

These are the results of the query
[query.zip](https://github.com/gnosygnu/xowa/files/3133892/query.zip)

On a separate note the first two results seem to be for the same image
Flag_of_the_United_States.svg
but they have different fil_ids 846 and 506149

**Comment by gnosygnu** [comment]
Thanks. That's a useful query to have.

I'd like to reduce disk space, primarily so it would make file upload / download easier. 5 GB is significant too.
